packages/Checkbox/src/checkbox.scss

```diff
@@ -1,291 +1,367 @@
-    top: 50%;
-    transform: translate(-50%, -50%);
-  }
-
-  &:checked:disabled::after {
-    animation: none;
-  }
-
-  &:disabled:hover {
-    border-color: var(--ids-checkbox-border);
-  }
+    appearance: none;
+    height: var(--ids-checkbox-size);
+    width: var(--ids-checkbox-size);
+    flex-shrink: 0;
+    margin: var(--ids-checkbox-top-space) 0 0;
+    display: inline-block;
+    position: relative;
+    cursor: pointer;
+    color: var(--ids-checkbox-check);
+    border: .1rem solid var(--ids-checkbox-border);
+    border-radius: var(--ids-checkbox-radius);
+    background: var(--ids-checkbox-background);
+    box-sizing: var(--ids-checkbox-box-sizing);
+    transition: background 0.05s ease-in-out;
+
+    &:hover{
+        border-color: var(--ids-checkbox-border-checked-hover);
+        background: var(--ids-checkbox-background-hover);
+    }
+
+    &:checked {
+        border-color: var(--ids-checkbox-border-checked);
+        background: var(--ids-checkbox-background-checked);
+    }
+
+    &:indeterminate {
+        border-color: var(--ids-checkbox-border-checked);
+        background: var(--ids-checkbox-background-checked);
+    }
+
+    &:disabled {
+        cursor: not-allowed;
+        border-color: var(--ids-checkbox-border-disabled);
+        background: var(--ids-checkbox-background-disabled);
+    }
+
+    &:focus {
+        outline: none;
+        box-shadow: var(--ids-checkbox-focus);
+    }
+
+    &:checked:hover,
+    &:indeterminate:hover {
+        border-color: var(--ids-checkbox-border-checked-hover);
+    }
+
+    &:focus:not(:focus-visible) {
+        box-shadow: none;
+    }
+
+    &:checked:disabled,
+    &:indeterminate:disabled {
+        background: var(--ids-checkbox-background-checked-disabled);
+        color: var(--ids-checkbox-check-color-disabled);
+    }
+
+    &:checked::after,
+    &:indeterminate::after {
+        content: '';
+        position: absolute;
+        color: var(--ids-checkbox-check-color);
+    }
+
+    &:checked:disabled::after,
+    &:indeterminate:disabled::after {
+        color: var(--ids-checkbox-check-color-disabled);
+    }
+
+    &:checked::after {
+        @extend %check;
+
+        display: block;
+        left: var(--ids-checkbox-check-left);
+        top: var(--ids-checkbox-check-top);
+        animation: check-animation ease-out 0.25s 0.15s both;
+    }
+
+    &:indeterminate::after {
+        @extend %indeterminate;
+
+        left: 50%;
+        top: 50%;
+        transform: translate(-50%, -50%);
+    }
+
+    &:checked:disabled::after {
+        animation: none;
+    }
+
+    &:disabled:hover {
+        border-color: var(--ids-checkbox-border-disabled);
+    }
 }
 
 .ids-checkbox--completion {
-  position: absolute;
-  clip: rect(.1rem, .1rem, .1rem, .1rem);
-  padding: 0;
-  border: 0;
-  height: .1rem;
-  width: .1rem;
-  overflow: hidden;
+    position: absolute;
+    clip: rect(.1rem, .1rem, .1rem, .1rem);
+    padding: 0;
+    border: 0;
+    height: .1rem;
```

Comment:
this won't compute the same between both themes as the rem is not on the same base.